traits/trait_types.py

```diff
@@ -1953,19 +1957,44 @@ def __init__(self, *args, **metadata):
                     "when using the 'values' keyword"
                 )
         else:
-            default_value = args[0]
-            if (len(args) == 1) and isinstance(default_value, EnumTypes):
-                args = default_value
-                default_value = enum_default(args)
-            elif (len(args) == 2) and isinstance(args[1], EnumTypes):
-                args = args[1]
+            if len(args) < 1:
+                raise TraitError("Enum trait requires at "
+                                 "least 1 argument.")
+
+            if len(args) == 1:
+                arg = args[0]
+                if isinstance(arg, EnumTypes):
+                    default_value = enum_default(arg)
+                    self.values = tuple(arg)
+
+                elif isinstance(arg, (str, bytes, bytearray)):
+                    default_value = arg
+                    self.values = {arg}
+
+                else:
+                    default_value = enum_default(arg)
+                    self.values = arg
+
+            elif len(args) == 2:
+                # If 2 args, the first is default, second is allowed values.
+                default_value = args[0]
+                allowed_vals = args[1]
+
+                if isinstance(allowed_vals, (str, bytes, bytearray)):
+                    self.values = tuple(args)
+                elif is_collection(allowed_vals):
+                    self.values = allowed_vals
+                else:
+                    self.values = tuple(args)
+            else:
+                default_value = args[0]
+                self.values = tuple(args)
 
             if isinstance(args, enum.EnumMeta):
```

Comment:
@corranwebster Do you know what this branch is for? As far as I can tell, `args` will always be a `tuple` here, so this branch is never exercised, and can be removed. (Possibly it ended up here as a result of a bad merge?)